File: elvitrina-backend/src/main/java/com/sudoers/elvitrinabackend/model/entity/Payment.java
package com.sudoers.elvitrinabackend.model.entity;

import com.sudoers.elvitrinabackend.model.enums.PaymentMethodType;
import com.sudoers.elvitrinabackend.model.enums.PaymentStatusType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private double amount;
    private LocalDateTime transactionDate;

    @Enumerated(EnumType.STRING)
    private PaymentMethodType method;

    @Enumerated(EnumType.STRING)
    private PaymentStatusType paystatus;

    @OneToMany(mappedBy = "payment", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<CustomOrder> orders;
}

File: elvitrina-frontend/src/app/core/models/Panier/payment.ts
import { PaymentMethodType } from "./PaymentMethodType.type";
import { PaymentStatusType } from "./PaymentStatusType.type";
export interface Payment {
  id: number;
  amount: number;
  transactionDate?: Date;
  method: PaymentMethodType;
  paystatus: PaymentStatusType;
  orderIds?: number[]; 
}
